python/llm_util.py

```diff
@@ -0,0 +1,153 @@
+                )
+
+                full_relationship = {
+                    Parameter.START.value: labels,
+                    Parameter.TYPE.value: relationship.type.name,
+                    Parameter.END.value: ":".join(target_labels),
+                }
+                if full_relationship not in self._all_relationships_list:
+                    self._all_relationships_list.append(full_relationship)
+
+                self._update_properties_dict(
+                    relationship,
+                    self._all_relationship_properties_dict,
+                    relationship.type.name,
+                )
+
+    def _get_raw_schema(self):
+        return {
+            Parameter.NODE_PROPS.value: self._all_node_properties_dict,
+            Parameter.REL_PROPS.value: self._all_relationship_properties_dict,
+            Parameter.RELATIONSHIPS.value: self._all_relationships_list,
+        }
+
+    def _get_prompt_ready_schema(self):
+        prompt_ready_schema = "Node properties are the following:\n"
+        for label in self._all_node_properties_dict.keys():
+            prompt_ready_schema += f"Node name: '{label}', Node properties: {self._all_node_properties_dict.get(label)}\n"
+
+        prompt_ready_schema += "\nRelationship properties are the following:\n"
+        for rel in self._all_relationship_properties_dict.keys():
+            prompt_ready_schema += f"Relationship Name: '{rel}', Relationship Properties: {self._all_relationship_properties_dict.get(rel)}\n"
+
+        prompt_ready_schema += "\nThe relationships are the following:\n"
+
+        for relationship in self._all_relationships_list:
+            prompt_ready_schema += f"['(:{relationship[Parameter.START.value]})-[:{relationship[Parameter.TYPE.value]}]->(:{relationship[Parameter.END.value]})']\n"
+
+        return prompt_ready_schema
+
+    def _update_properties_dict(self, graph_object, all_properties_dict, key):
+        for property_name in graph_object.properties.keys():
+            if not all_properties_dict.get(key):
+                all_properties_dict[key] = [
+                    {
+                        Parameter.PROPERTY.value: property_name,
+                        Parameter.TYPE.value: type(
+                            graph_object.properties.get(property_name)
+                        ).__name__,
+                    }
+                ]
+            else:
+                if property_name not in [
+                    d[Parameter.PROPERTY.value] for d in all_properties_dict[key]
+                ]:
+                    all_properties_dict[key].append(
+                        {
+                            Parameter.PROPERTY.value: property_name,
+                            Parameter.TYPE.value: type(
+                                graph_object.properties.get(property_name)
+                            ).__name__,
+                        }
+                    )
```

Comment:
```suggestion
        for property_name in graph_object.properties.keys():
            if not all_properties_dict.get(key):
                all_properties_dict[key] = [
                    {
                        Parameter.PROPERTY.value: property_name,
                        Parameter.TYPE.value: type(
                            graph_object.properties.get(property_name)
                        ).__name__,
                    }
                ]
                return
            
             if property_name  in [
                  d[Parameter.PROPERTY.value] for d in all_properties_dict[key]
              ]: 
               return
              all_properties_dict[key].append(
                  {
                      Parameter.PROPERTY.value: property_name,
                      Parameter.TYPE.value: type(
                          graph_object.properties.get(property_name)
                      ).__name__,
                  }
              )
```

Reply:
The general suggestion here is to reduce indent as much as possible. I think with early returns you can do that here

Reply:
I think that `continue` should be used here, but I get your point. Thanks